### PR TITLE
RyuJIT: Fold Popcnt.PopCount with constant argument

### PR DESCRIPTION
**Pull request by EgorBo**
Since [BitOperations.IsPow2](https://github.com/dotnet/runtime/issues/31297) API is approved and [looks like](https://github.com/dotnet/runtime/pull/36163) it's going to use PopCount under the hood I think it makes sense to optimze PopCount for constant arguments (or if the argument becomes a constant after inlining) thus `IsPow2` will be folded for such arguments as well.

Example:
```csharp
int Test() => Foo(15);

int Foo(uint x) => BitOperations.PopCount(x);
```

Codegen for `Test` with this PR:
```asm
G_M38134_IG01:
G_M38134_IG02:
       B804000000           mov      eax, 4
G_M38134_IG03:
       C3                   ret      
; Total bytes of code: 6
```

/cc @tannergooding @john-h-k

### CODE REVIEW

File: src/coreclr/src/jit/importer.cpp [hunk 1]
```diff
@@ -3520,6 +3520,18 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
             {
                 GenTree* hwintrinsic = impHWIntrinsic(ni, clsHnd, method, sig, mustExpand);
 
+#ifdef TARGET_XARCH
```
Comment (tannergooding) [inline src/coreclr/src/jit/importer.cpp:3523]: The same should be valid for ARM64, right?

The actual hardware intrinsic is a bit different, but that shouldn't stop constant folder for `BitOperations`

**Comment by jkotas** [comment]
Would it better to optimize the BitOperations in platform-neutral way?

**Comment by EgorBo** [comment]
@tannergooding @jkotas ok, changed to BitOperations.PopCount
in theory can do the same for `LeadingZeroCount` and `TrailingZeroCount`

**Comment by EgorBo** [comment]
@tannergooding any idea why so many failing tests?
all of them are
```
Method System.Numerics.Vector`1[]:Max was compiled but should not have been
```
I recall you've fixed something similar recently

**Comment by tannergooding** [comment]
@EgorBo, it's because you added a new check for `System.Numerics` which conflicts with the existing check under the `#ifdef FEATURE_HW_INTRINSICS` block: <https://github.com/dotnet/runtime/pull/37836/files#diff-b8d003a58643e5595d2920ca5993b952L4506>

**Comment by EgorBo** [comment]
> @EgorBo, it's because you added a new check for `System.Numerics` which conflicts with the existing check under the `#ifdef FEATURE_HW_INTRINSICS` block: https://github.com/dotnet/runtime/pull/37836/files#diff-b8d003a58643e5595d2920ca5993b952L4506

Ouch, thanks!

**Comment by AndyAyersMS** [comment]
@EgorBo @tannergooding is this ready to merge?

**Comment by EgorBo** [comment]
It's finished from my end

**Review by AndyAyersMS — APPROVED** [review]
Looks good.

**Comment by AndyAyersMS** [comment]
@EgorBo Thank you for your steady stream of improvements to the jit.

**Pull request merged**